### PR TITLE
Final endpoints

### PR DESCRIPTION
**Pull request by tvaroglu**
- Changes Implemented:
  - Added DELETE `/api/v1/users/:user_id/gym_memberships/:gym_membership_id/events/:id`
  - Added helper methods (and endpoints) to retrieve user's friends at the same gym (`api/v1/users/:user_id/friendships#index`) and users at the same gym the user is NOT already friends with (`api/v1/users/:user_id/gym_members`)
  - Added `meta` to `/api/v1/users/:user_id/events#index` to return `user.full_name` along with all events, to facilitate dashboard view

- Quality Control Checklist:

  - [x] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
    - Still some cleanup to be done, but only a few infractions (i.e. a couple lines too long).  Assuming we may comb back through all this anyways, but happy to refactor live, as we are merging this PR, if that is preferred.
  - [x] 100% SimpleCov test coverage (if below, please clarify below)
  - [x] Last Commit passes TravisBuild

- Blockers (if applicable):

- Next Steps & Additional Notes:

### CODE REVIEW

File: app/controllers/api/v1/users/events_controller.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ def index
     # events (e.g. if we decide to add activity log to dashboard)
     events = user.upcoming_events
 
-    render json: EventSerializer.new(events).serializable_hash, status: :ok
+    options = {}
+    options[:include] = [:user]
+    render json: EventSerializer.new(events, options).serializable_hash, status: :ok
```
Comment (Scott-Borecki) [inline app/controllers/api/v1/users/events_controller.rb:12]: Love this!

**Review by ealwafai — APPROVED** [review]
Looks great! I just need to figure out the Proc.new

File: app/controllers/api/v1/users/find_controller.rb [hunk 2]
```diff
@@ -2,9 +2,7 @@
 # helpers.
 class Api::V1::Users::FindController < ApplicationController
   def index
-    if params[:google_id].blank?
-      return render json: error_message, status: :not_found
-    end
+    return error_message if params[:google_id].blank?
 
     user = User.find_by!(google_id: params[:google_id])
```
Comment (Scott-Borecki) [inline app/controllers/api/v1/users/find_controller.rb:7]: Love this refactor.

File: app/controllers/api/v1/users/gym_members_controller.rb [hunk 3]
```diff
@@ -2,22 +2,23 @@
...
 class Api::V1::Users::GymMembersController < ApplicationController
   def index
-    if params[:yelp_gym_id].blank?
-      return render json: error_message, status: :not_found
-    end
+    return error_message if params[:yelp_gym_id].blank?
 
     user = User.find(params[:user_id])
-    users = user.non_friends_at_same_gym(params[:yelp_gym_id]) if user
+    users = user.non_friends_at_same_gym(params[:yelp_gym_id])
```
Comment (Scott-Borecki) [inline app/controllers/api/v1/users/gym_members_controller.rb]: Love this refactor

**Review by Scott-Borecki — APPROVED** [review]
Looks great!  Nice work!

**Pull request merged**